flower/utils/pg_storage.py

```diff
@@ -60,16 +61,29 @@ def open_connection(user, password, database, host, port, use_ssl):
         host=host, port=port, ssl=use_ssl
     )
 
-    # Create schema if database is empty
+    # Create schema if table is missing
     cursor = connection.cursor()
     try:
         cursor.execute(_all_tables)
-        tables = cursor.fetchone()
+        tables = cursor.fetchall()
         if tables is None:
             logger.debug('Database empty, executing schema definition.')
             for statement in _schema:
                 cursor.execute(statement)
             connection.commit()
+
+        table_match = False
```

Comment:
To make this more concise and to avoid duplicating logic for executing schema definition, consider changing the above condition to something like:
```python
if tables is None or not any(('events' in table[2]) for table in tables):
    ...
```

The logger could accommodate both situations by logging `Table "events" missing, executing schema definition.`